### ISSUE TITLE
ERR Extension 'ms-python.debugpy CANNOT USE these API proposals 'portsAttributes, contribIssueReporter,

### ISSUE THREAD

**Issue by grigio** [microsoft/vscode-python-debugger#431]
PS: I tried also the integrated bug report feature but it doesn't send

# Behaviour

The extension doesn't load

```
Electron Security Warning (enableBlinkFeatures) This renderer process has additional "enableBlinkFeatures"
  enabled. This exposes users of this app to some security risk. If you do not
  need this feature, you should disable it.

For more information and help, consult
https://electronjs.org/docs/tutorial/security.
This warning will not show up
once the app is packaged.
(anonymous) @ VM4 sandbox_bundle:2
log.ts:419  INFO Started local extension host with pid 105572.
log.ts:439   ERR Extension 'ms-python.debugpy CANNOT USE these API proposals 'portsAttributes, contribIssueReporter, debugVisualization'. You MUST start in extension development mode or use the --enable-proposed-api command line flag
log.ts:439   ERR Extension 'ms-python.python CANNOT USE these API proposals 'contribEditorContentMenu, quickPickSortByLabel, testObserver, quickPickItemTooltip, terminalDataWriteEvent, terminalExecuteCommandEvent, contribIssueReporter, terminalShellIntegration'. You MUST start in extension development mode or use the --enable-proposed-api command line flag
log.ts:439   ERR Extension 'ms-toolsai.jupyter CANNOT USE these API proposals 'notebookDeprecated, notebookMessaging, notebookMime, contribNotebookStaticPreloads, portsAttributes, quickPickSortByLabel, notebookKernelSource, interactiveWindow, quickPickItemTooltip, notebookExecution, notebookCellExecution, notebookVariableProvider'. You MUST start in extension development mode or use the --enable-proposed-api command line flag
log.ts:439   ERR Extension 'ms-toolsai.jupyter-renderers CANNOT USE these API proposals 'contribNotebookStaticPreloads'. You MUST start in extension development mode or use the --enable-proposed-api command line flag
log.ts:439   ERR [ms-python.debugpy]: issue/reporter is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["contribIssueReporter"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api ms-python.debugpy
log.ts:439   ERR [ms-python.python]: issue/reporter is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["contribIssueReporter"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api ms-python.python
log.ts:439   ERR [ms-python.python]: editor/content is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["contribEditorContentMenu"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api ms-python.python
webviewElement.ts:482 An iframe which has both allow-scripts and allow-same-origin for its sandbox attribute can escape its sandboxing.
```

## Steps to reproduce:

1. XXX

# Diagnostic data

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
XXX
```

</p>
</details>

**Comment by paulacamargo25** [comment]
Thanks for your bug report, i can't repro the error, can you try again? but make sure that you have the last version of vscode, Python Extension, and Python Debugger Extension? Thanks

**Comment by grigio** [comment]
I installed it via Archlinux
![image](https://github.com/user-attachments/assets/7db99070-4d86-45f4-99dd-f03db2539259)

i tried to clean config and extensions and it persists

**Comment by grigio** [comment]
![image](https://github.com/user-attachments/assets/ed3e92de-2dd4-4e55-9e69-cef85e47fc81)

Ok, sorry with official visual studio code it works

**Comment by hawav** [comment]
> I installed it via Archlinux ![image](https://github.com/user-attachments/assets/7db99070-4d86-45f4-99dd-f03db2539259)
> 
> i tried to clean config and extensions and it persists

I encounter the same problem. Why doesn't it work on Code-OSS?

**Comment by eljonny** [comment]
I just noticed this starting to happen recently, very weird didn't see this before.

I don't want to stop using Code-OSS on Arch Linux, so I did some digging, and found [this comment](https://github.com/microsoft/vscode-python/issues/24013#issuecomment-2330559040) on a [closed issue](https://github.com/microsoft/vscode-python/issues/24013) of similar nature on the [vscode-python](microsoft/vscode-python) repository.

It looks like there's a missing top-level property in the code-oss `product.json` file, at `/usr/lib/code/product.json` in my installation.
The API proposals that are missing, in my specific case, are:

- `portsAttributes`
- `debugVisualization`

I added the following property key and value to `product.json`:

```json
"extensionEnabledApiProposals": {
	"ms-python.debugpy": [
		"portsAttributes",
		"debugVisualization"
	]
}
```

Then restarted Code-OSS, then re-installed the extension, and it is working without error now.
